lib/install/utils.go

```diff
@@ -336,7 +342,7 @@ func (r ProgressPoller) Run(ctx context.Context) error {
 				r.WithError(err).Warn("Failed to query operation progress.")
 				continue
 			}
-			if lastProgress != nil && lastProgress.IsEqual(*progress) {
+			if shouldIgnoreProgress(progress, lastProgress) {
```

Comment:
This fixes an issue with resuming after an error in which case the agent/installer would show the last error as a new progress entry which was confusing.